sideboard/lib/_profiler.py

```diff
@@ -0,0 +1,271 @@
+ * Uses `cProfile` instead of `profile` for better performance.
+
+Profiling data can be collected using the @profile decorator on functions and
+methods. The profiling results can be viewed at http://servername/profile/.
+
+Good candidates for profiling are the outermost functions that generate your
+web pages, usually exposed as cherrypy endpoints via @cherrypy.expose::
+
+    import cherrypy
+    from sideboard.lib import profile
+
+    class Root(object):
+        @cherrypy.expose
+        @profile
+        def index(self):
+            # Create and return the index page
+            return '<html/>'
+
+
+But any regular function can be profiled using the @profile decorator::
+
+    from sideboard.lib import profile
+
+    @profile
+    def some_interesting_function():
+       # Do some stuff
+
+
+The following config options control how the profiler operates, see
+configspec.ini for more details::
+
+    [cherrypy]
+    profiling.on = boolean(default=True)
+    profiling.path = string(default="%(root)s/data/profiler")
+    profiling.aggregate = boolean(default=False)
+    profiling.strip_dirs = boolean(default=False)
+
+"""
+import io
+import os
+import os.path
+import cProfile
+import pstats
+from datetime import datetime
+from functools import wraps
+from glob import glob
+
+import cherrypy
+from sideboard.config import config
+from sideboard.lib import entry_point, listify
+
+
+def _new_func_strip_path(func_name):
+    """
+    Adds the parent module to profiler output for `__init__.py` files.
+
+    Copied verbatim from cherrypy/lib/profiler.py.
+    """
+    filename, line, name = func_name
+    if filename.endswith('__init__.py'):
+        return os.path.basename(filename[:-12]) + filename[-12:], line, name
+    return os.path.basename(filename), line, name
+
+pstats.func_strip_path = _new_func_strip_path
+
+
+@entry_point
+def cleanup_profiler():
+    """
+    Deletes all `*.prof` files in the profiler's data directory.
+
+    This is useful when you've created tons of profile files that you're no
+    longer interested in. Exposed as a `sep` command::
+
+        $ sep cleanup_profiler
+
+    The profiler directory is specified in the config by::
+
+        [cherrypy]
+        profiling.path = 'path/to/profile/data'
+
+    """
+    profiling_path = config['cherrypy'].get('profiling.path', None)
+    if profiling_path:
+        for f in glob(os.path.join(profiling_path, '*.prof')):
+            os.remove(f)
+
+
+def profile(func):
+    """
+    Decorator to capture profile data from a method or function.
+
+    If profiling is disabled then this decorator is a no-op, and the original
+    function is returned unmodified. Since the original function is returned,
+    this decorator does not incur any performance penalty if profiling is
+    disabled. To enable or disable profiling use the following setting in your
+    config::
+
+            [cherrypy]
+            profiling.on = 'True'  # Or 'False' to disable
```

Comment:
Two minor style nitpicks:

1) Why are you indenting 12 spaces instead of 8 spaces here?

2) ConfigObj allows (and I believe prefers) unquoted True/False for boolean values, so you can just say ``profiling.on = True``.